object-path-immutable.d.ts

```diff
@@ -5,6 +5,7 @@ interface WrappedObject<T> {
     push(path?: Path, value?: any): WrappedObject<T>
     del(path?: Path): WrappedObject<T>
     assign(path?: Path, source?: T): WrappedObject<T>
+    merge(path?: Path, source?: T): WrappedObject<T>
```

Comment:
```suggestion
    merge(path?: Path, source?: any): WrappedObject<T>
```
Needs to be `any` to support nested assignment.